src/Microsoft.ML.TimeSeries/SsaSpikeDetector.cs

```diff
@@ -133,6 +134,15 @@ private static IDataTransform Create(IHostEnvironment env, ModelLoadContext ctx,
             return new SsaSpikeDetector(env, ctx).MakeDataTransform(input);
         }
 
+        public override IStatefulTransformer Clone()
+        {
+            var clone = (SsaSpikeDetector)MemberwiseClone();
+            clone.Model = clone.Model.Clone();
+            clone.StateRef = (State)clone.StateRef.Clone();
```

Comment:
Do we have 1 to 1 correspondence between Model and State now? if so maybe you could make a State a part of a Model now? So State object is tracked inside the Model

Reply:
May be we can merge State and Model in one but I will prefer doing this after part of bug-bash because its more of a cleanup and I'm short on time for check-in.

---
In reply to: [236525220](https://github.com/dotnet/machinelearning/pull/1727#discussion_r236525220) [](ancestors = 236525220)